src/commands/deploy.rs

```diff
@@ -640,43 +644,39 @@ async fn check_healthz(url: &str) -> Result<()> {
 
 const READINESS_POLL_INTERVAL_SECS: u64 = 2;
 
-async fn wait_for_ready(
-    app_domain: &str,
-    hippo_url: &str,
-    cfg: &spin_loader::local::config::RawAppManifest,
-    readiness_timeout_secs: u16,
-) {
+async fn wait_for_ready(app_base_url: &Url, bindle_version: &str, readiness_timeout_secs: u16) {
     if readiness_timeout_secs == 0 {
         return;
     }
 
-    if cfg.components.is_empty() {
-        return;
-    }
-
-    let url_result = Url::parse(hippo_url);
-    let scheme = match &url_result {
-        Ok(url) => url.scheme(),
-        Err(_) => "http",
-    };
-
-    let path = WELL_KNOWN_HEALTH_PATH;
-    let health_check_url = format!("{}://{}{}", scheme, app_domain, path);
+    let app_info_url = app_base_url
+        .join(WELL_KNOWN_PREFIX.trim_start_matches('/'))
```

Comment:
I'm a bit puzzled that you diligently add the trailing slash in `parse_url` "so join works" and then need to remove the leading slash of `WELL_KNOWN_PREFIX` to compensate for that - it makes me wonder why not just declare WKP without the leading slash.  Is it because other consumers of WKP presume that the URL does _not_ have the trailing slash?

Reply:
> Is it because other consumers of WKP presume that the URL does not have the trailing slash?

Yes, it is used directly in a `.strip_prefix(WELL_KNOWN_PREFIX)` in `spin-http`. The code is either awkward here or awkward there.